### PR TITLE
Load llama ckpt to meta to work around OOM issue on CPU

### PR DESCRIPTION
**Pull request by mandy-li**
When generate text with DeepSpeed library, this PR loads llama ckpt to meta to work around OOM issue on CPU.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: examples/text-generation/checkpoint_utils.py [hunk 1]
```diff
@@ -75,11 +75,11 @@ def write_checkpoints_json(model_name_or_path, local_rank, checkpoints_json):
...
 
 
-def model_is_bloom(config):
+def model_on_meta(config):
     """
-    Checks if the given config belongs to a BLOOM-like model.
+    Checks if load the model to meta.
     """
-    return config.model_type == "bloom"
+    return config.model_type == "bloom" or config.model_type == "llama"
```
Comment (ZhaiFeiyue) [inline examples/text-generation/checkpoint_utils.py:82]:
```suggestion
    return config.model_type in ["bloom", "llama"]
```

**Comment by ZhaiFeiyue** [comment]
@mandy-li code style check failed, please run `make style`

**Review by ZhaiFeiyue — APPROVED** [review]
LGTM

**Comment by regisss** [comment]
@mandy-li I tried running the following command
```
python ../gaudi_spawn.py --use_deepspeed --world_size 8 run_generation.py --model_name_or_path meta-llama/Llama-2-70b-hf --batch_size 1 --use_hpu_graphs --use_kv_cache --max_new_tokens 100
```
but got this error
```
RuntimeError: Tensor on device hpu:0 is not on the expected device meta!
```

Which command did you use to run the model?

**Comment by LeoZhao-Habana** [comment]
need install Mandy version deepspeed, pip install git+https://github.com/mandy-li/DeepSpeed.git
otherwise the result is wrong.

**Comment by regisss** [comment]
> need install Mandy version deepspeed, pip install git+https://github.com/mandy-li/DeepSpeed.git otherwise the result is wrong.

@mandy-li Have you also observed some differences between your fork and Habana's?

**Comment by mandy-li** [comment]
@regisss , from my testing, no. This is why i asked you to merge

**Comment by LeoZhao-Habana** [comment]
By my test on LLAMA2 70B, this is commit needed for correct result, https://github.com/mandy-li/DeepSpeed/commit/672885c3dac731efd87ba2d3db6bd1d8ffca3ac4     which is missed or not merged on habana deepspeed main branch.

**Comment by ZhaiFeiyue** [comment]
@mandy-li have you tested with a clean env(docker)?

**Comment by LeoZhao-Habana** [comment]
this issue happens on Gaudi1 only. patch is still needed, and need to check rootcause of  difference btw gaudi1 and gaudi2

**Pull request merged**